### PR TITLE
`:has()` pseudo : more docs, tests and add typings

### PR DESCRIPTION
**Pull request by romainmenke**
- add docs for all plugin options
- explain how CORS comes into play with the browser plugin
- change tests to trigger CORS (so that we know our docs are correct)
- add typescript for the plugin
- get stryker coverage back up to 100% for everything in `src/encode/`
- investigated and added tests for plugin ordering.

-----------

see : https://github.com/csstools/postcss-plugins/issues/100#issuecomment-1007800947

After this change is merged and released we wait until Safari ships `:has`.
If we receive no further feedback, this version might become the next major update for `css-has-pseudo`.

### CODE REVIEW

File: experimental/css-has-pseudo/README-BROWSER.md [hunk 1]
```diff
@@ -40,8 +40,72 @@ cssHasPseudo(document);
 <script>cssHasPseudo(document)</script>
 ```
 
+⚠️ Please use a versioned url, like this : `https://unpkg.com/@csstools/css-has-pseudo-experimental@0.1.0/dist/browser-global.js`
```
Comment (Antonio-Laguna) [inline experimental/css-has-pseudo/README-BROWSER.md:43]:
```suggestion
⚠️ Please use a versioned url, like this : `https://unpkg.com/@csstools/css-has-pseudo-experimental@0.1.2/dist/browser-global.js`
```

Let's use latest

**Review by Antonio-Laguna — APPROVED** [review]
This is impressive! Loved the TS support and making this ready to swap with major.

Left a minor comment but happy to leave as is, your call :)

File: experimental/css-has-pseudo/README-BROWSER.md (hunk 1 above)
Comment (romainmenke) [reply experimental/css-has-pseudo/README-BROWSER.md:43]: This should also be generated :)
Made a note for this.

File: experimental/css-has-pseudo/README-BROWSER.md [hunk 2]
```diff
@@ -40,8 +40,72 @@ cssHasPseudo(document);
 <script>cssHasPseudo(document)</script>
 ```
 
+⚠️ Please use a versioned url, like this : `https://unpkg.com/@csstools/css-has-pseudo-experimental@0.1.2/dist/browser-global.js`
```
Comment (romainmenke) [inline experimental/css-has-pseudo/README-BROWSER.md:43]:
```suggestion
⚠️ Please use a versioned url, like this : `https://unpkg.com/@csstools/css-has-pseudo-experimental@0.2.0/dist/browser-global.js`
```

Next release will be 0.2.0

**Comment by romainmenke** [comment]
Thank you for reviewing all this!

**Pull request merged**